astropy/convolution/convolve.py

```diff
@@ -1001,7 +1001,9 @@ def interpolate_replace_nans(array, kernel, convolve=convolve, **kwargs):
     return newarray
 
 
-def convolve_models(model, kernel, mode="convolve_fft", **kwargs):
+def convolve_models(
+    model, kernel, bounding_box, resolution, mode="convolve_fft", cache=True, **kwargs
```

Comment:
This is API change, so maybe a second change log entry to explain this change?

Reply:
In order to preserve backwards compatibility, with the API this probably needs to actually be:

```python
def convolve_models(
    model, kernel, mode="convolve_fft", bounding_box=None, resolution=None, cache=True, **kwargs
)
```
where the `None` values map to  assumed defaults. It might be reasonable to issue a warning if they are not specified.

This way no current code which positionally passes the `mode` keyword and/or fails to define the `bounding_box` or `resolution` keywords will be immediately broken.

Reply:
Since `resolution`, `cache`, and `bounding_box` have never been allowed as positional arguments, an ever more conservative signature would be
```python
def convolve_models(
    model, kernel, mode="convolve_fft", *, bounding_box=None, resolution=None, cache=True, **kwargs
)
```

Reply:
@WilliamJamieson @neutrinoceros 
I will make the recommended change to the `convolve_models()` signature, however I will point out that this only preserves backward compatibility in the API. The functionality has changed such that any calls to the function will undoubtedly need to be updated, adding the new keywords, in order to restore the previous behavior. In fact, making the API backward compatible may obfuscate the issue because users are likely to get a `ValueError: One of the requested xi is out of bounds` error when the underlying interpolation fails because the bounding box has not been properly defined.

Reply:
I have made the requested change, but have to say that I don't like it. It seems buggy to implement default values for `bounding_box` and `resolution` that may not match the dimensionality of the input model and kernel. Maybe I should just rollback the changes to the `convolve_models` function and improve the documentation to note that that function does not behave like one would expect but instead just convolves the output of `model` and `kernel`.

Reply:
> Maybe I should just rollback the changes

If you are up for it, does not hurt to have a competing PR with just the doc changes, and see which one gets merged. If you do that, you can specify that the new PR would close this PR, to be clear. Thanks!

Reply:
I cannot unresolved this thread but I think it was prematurely closed. I'll open a other one so I'm sure it's visible.

Reply:
I unresolved it for you.